Source/JavaScriptCore/wasm/WasmOperationsInlines.h

```diff
@@ -312,6 +311,122 @@ inline void arraySet(Instance* instance, uint32_t typeIndex, EncodedJSValue arra
     arrayObject->set(index, value);
 }
 
+inline bool arrayFill(Instance* instance, uint32_t typeIndex, EncodedJSValue arrayValue, uint32_t offset, EncodedJSValue value, uint32_t size)
+{
+#if ASSERT_ENABLED
+    ASSERT(typeIndex < instance->module().moduleInformation().typeCount());
+    const Wasm::TypeDefinition& arraySignature = instance->module().moduleInformation().typeSignatures[typeIndex]->expand();
+    ASSERT(arraySignature.is<ArrayType>());
+#else
+    UNUSED_PARAM(instance);
+    UNUSED_PARAM(typeIndex);
+#endif
+
+    JSValue arrayRef = JSValue::decode(arrayValue);
+    ASSERT(arrayRef.isObject());
+    JSWebAssemblyArray* arrayObject = jsCast<JSWebAssemblyArray*>(arrayRef.getObject());
+
+    CheckedUint32 lastElementIndexChecked = offset;
+    lastElementIndexChecked += size;
+
+    if (lastElementIndexChecked.hasOverflowed())
+        return false;
+
+    if (lastElementIndexChecked > arrayObject->size())
+        return false;
+
+    arrayObject->fill(offset, value, size);
+    return true;
+}
+
+// FIXME:
+// To be consistent it would make sense to include dst and src type indices, but they
+// are not necessary for operation and hits a limitation of BBQ JIT calls.
```

Comment:
This comment is about working around what might be a bug in BBQJIT. I'm planning to file a proper bug report and put the bug link in here.

The basic issue seems to be that for a call to a WasmOperation, if the call arguments need to be allocated on the stack, it breaks in debug build mode on Linux. This wasn't an issue before because I don't think other operations exceed 6 arguments (x86-64 unix calling convention has 6 arg regs).